docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-enable-cluster-label-verification/_index.md

```diff
@@ -0,0 +1,96 @@
+
+In this document we will describe the step on how to migrate a Mimir installation to prevent two separate memberlist gossip hash ring to join into one.
+
+There are three steps of the migration which we will describe in details in the migration section. But in brief summary the steps are:
+
+1. Disable memberlist cluster label verification
+1. Set cluster label to all Mimir components
+1. Enable memberlist cluster label verification again
+
+The migration should take around 30 minutes. The risk of not doing this migration if you have Mimir and Tempo or Loki in the same cluster is the possibility of those different backend to talk with each other.
+
+## How The Migration Solve The Issue
+
+Non-Mimir component can merge with Mimir memberlist because by default, memberlist is not namespaced and applies globally. As an example, consider a Loki pod is terminated and the IP address reused by a new Mimir pod in the same cluster. This might cause other Loki components try to talk and sending data to this new Mimir pod.
+
+Cluster label will prevent such situation by only allowing communication for components that has same cluster label. Once that is enabled, before memberlist try to communicate with other components, it will verify whether that new components are having same cluster label and only allowing memberlist to communicate if the parties are having the same cluster label.
+
+## Migration
+
+### 1. Disable memberlist cluster label verification
+
+Cluster label verification flag is enabled by default. However, cluster label default value is an empty string. If different systems that using memberlist and not setting the cluster label, they can talk with each other. To disable cluster label verification flag, set the following structured config.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: true
+```
+
+Rollout the installation to apply the configuration changes by running `helm upgrade mimir-distributed -f values.yaml`.
+
+### 2. Set cluster label to all Mimir components
+
+Set cluster label to all Mimir components by setting the following configuration. Once the configuration is applied, all Mimir components will only communicate via memberlist if the other component is having the same cluster label.
+
+```
+mimir
+  structuredConfig:
+    memberlist:
+      cluster_label_verification_disabled: true
+      cluster_label: "$helm-release-name"
+```
+
+Apply the configuration changes by running `helm upgrade mimir-distributed -f values.yaml`.
+
+### 3. Enable memberlist cluster label verification
+
+Set `memberlist.cluster_label_verification_disabled` to false, to re-enable memberlist cluster label verification.
```

Comment:
technically the path is `mimir.structuredConfig.memberlist.cluster_label_verification_disabled`

Reply:
also can't we just remove the key from the config? That way users don't need to have some line in their config "forever"

Reply:
Removed the config and mentioned it in the text that the removal of the key config means re-enabling cluster label verification with the new cluster label.